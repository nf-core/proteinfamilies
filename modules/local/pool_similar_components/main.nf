process POOL_SIMILAR_COMPONENTS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/64/64288a2d9e2c21c4f3cd9f8cb8476c04d323de98dbd8a310265077566f9bb75a/data' :
        'community.wave.seqera.io/library/networkx_pandas_python:ab2ee9a9e2c80a69' }"

    input:
    tuple val(meta), path(similarities)

    output:
    tuple val(meta), path("pooled_components.txt"), emit: pooled_components
    path "versions.yml"                           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    """
    pool_similar_components.py \\
        --input_csv ${similarities} \\
        --out_file pooled_components.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        pandas: \$(python -c "import importlib.metadata; print(importlib.metadata.version('pandas'))")
        networkx: \$(python -c "import importlib.metadata; print(importlib.metadata.version('networkx'))")
    END_VERSIONS
    """

    stub:
    """
    touch pooled_components.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        pandas: \$(python -c "import importlib.metadata; print(importlib.metadata.version('pandas'))")
        networkx: \$(python -c "import importlib.metadata; print(importlib.metadata.version('networkx'))")
    END_VERSIONS
    """
}
