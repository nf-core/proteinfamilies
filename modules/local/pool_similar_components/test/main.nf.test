nextflow_process {

    name "Test Process POOL_SIMILAR_COMPONENTS"
    script "../main.nf"
    process "POOL_SIMILAR_COMPONENTS"

    test("proteinfamilies - csv") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.pipelines_testdata_base_path + 'test_data/modules/similarities.csv', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.pooled_components,
                    process.out.versions.collect { path(it).yaml }
                    ).match()}
            )
        }
    }

    test("proteinfamilies - csv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.pipelines_testdata_base_path + 'test_data/modules/similarities.csv', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.pooled_components,
                    process.out.versions.collect { path(it).yaml }
                    ).match()}
            )
        }
    }
}
