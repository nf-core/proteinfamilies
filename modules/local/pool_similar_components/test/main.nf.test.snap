{
    "proteinfamilies - csv": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "pooled_components.txt:md5,784c1d263424fbd0d225308f9a719feb"
                ]
            ],
            [
                {
                    "POOL_SIMILAR_COMPONENTS": {
                        "python": "3.13.1",
                        "pandas": "2.3.3",
                        "networkx": 3.5
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-09T11:12:44.309501549"
    },
    "proteinfamilies - csv - stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "pooled_components.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                {
                    "POOL_SIMILAR_COMPONENTS": {
                        "python": "3.13.1",
                        "pandas": "2.3.3",
                        "networkx": 3.5
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-09T11:12:49.694151275"
    }
}